src/telemetry.h

```diff
@@ -85,6 +85,19 @@ void tm_set_config_file(char *c_file);
 int tm_create_record(struct telem_ref **t_ref, uint32_t severity,
                      char *classification, uint32_t payload_version);
 
+/**
+ * Sets the event_id to a user defined id
+ *
+ * @param t_ref A pointer to a telem_ref struct that was previously initialized
+ *        this struct will have an event_id already initialized and it will be
+ *        overriden by a user provided event_id
+ * @param event_id A pointer to a string that contains the 32 character length
+ *        event_id
+ *
+ * @return 0 on success, or s negative errno-style value on error
```

Comment:
If I'm reading the functions right, this one always returns 0 or 1.

Reply:
I'm reading it this way too. @alexjch Should the function be returning a negative value, presuming that other public API functions do the same?